ha/setup/ha_setup.py

```diff
@@ -0,0 +1,315 @@
+        Print usage instructions
+        """
+        sys.stderr.write(
+            f"usage: {prog} [-h] <cmd> <--config url> [--plan <TYPE>] <args>...\n"
+            f"where:\n"
+            f"cmd   post_install, config, init, test, upgrade, reset, cleanup, backup, restore\n"
+            f"--config   Config URL\n"
+            f"--plan   Test plan")
+
+    @staticmethod
+    def get_command(desc: str, argv: dict):
+        """
+        Return the Command after parsing the command line.
+        """
+        parser = argparse.ArgumentParser(desc)
+
+        subparsers = parser.add_subparsers()
+        cmds = inspect.getmembers(sys.modules[__name__])
+        cmds = [(x, y) for x, y in cmds
+            if x.endswith("Cmd") and x != "Cmd"]
+        for name, cmd in cmds:
+            cmd.add_args(subparsers, cmd, name)
+        args = parser.parse_args(argv)
+        return args.command(args)
+
+    @staticmethod
+    def add_args(parser: str, cls: str, name: str):
+        """
+        Add Command args for parsing.
+        """
+        parser1 = parser.add_parser(cls.name, help='setup %s' % name)
+        parser1.add_argument('--config', help='Config URL')
+        parser1.add_argument('--plan', nargs="?", default=[], help='Test plan')
+        parser1.add_argument('args', nargs='*', default=[], help='args')
+        parser1.set_defaults(command=cls)
+
+class PostInstallCmd(Cmd):
+    """
+    PostInstall Setup Cmd
+    """
+    name = "post_install"
+
+    def __init__(self, args: dict):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process post_install command.
+        """
+
+        # Create a directory and copy config file
+        if os.path.exists(const.HA_CONFIG_FILE):
+            os.remove(const.HA_CONFIG_FILE)
+        os.makedirs(const.CONFIG_DIR, exist_ok=True)
+        shutil.copyfile(const.SOURCE_CONFIG_FILE, const.HA_CONFIG_FILE)
+        Log.info("Copied HA config file.")
+        # Pre-requisite checks are done here.
+        # Make sure the pacemaker, corosync and pcs packages have been installed
+        PkgV().validate('rpms', const.PCS_CLUSTER_PACKAGES)
+
+class ConfigCmd(Cmd):
+    """
+    Setup Config Cmd
+    """
+    name = "config"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process config command.
+        """
+        # Read machine-id and using machine-id read minion name from confstore
+        # This minion name will be used for adding the node to the cluster.
+        nodelist = []
+        command = "cat /etc/machine-id"
+        machine_id, err, rc = self._execute.run_cmd(command, check_error=True)
+        Log.info(f"Read machine-id. Output: {machine_id}, Err: {err}, RC: {rc}")
+        minion_name = Conf.get(self._index, f"cluster.server_nodes.{machine_id.strip()}")
+        nodelist.append(minion_name)
```

Comment:
Minion_name may not resolve to an IP. In R1, srvnode-1 resolved to private IP but in R2, that might not be the case. We should get it confirmed by Provisioner team.

Reply:
Minion_name is just the variable name we are using here, otherwise I believe we are referring to the right key/value in the confstore.

Reply:
Lets resolve it during integration with provisioner.